syntax = "proto3";

package complaint_service;

import "google/protobuf/empty.proto";

option go_package = "github.com/milkyhoop/template-service-golang/internal/delivery/pb/template;template";
// option python_package = "app.protos";

service Complaint_service {
  rpc DoSomething (Complaint_serviceRequest) returns (Complaint_serviceResponse);
  rpc CreateComplaint (CreateComplaintRequest) returns (CreateComplaintResponse); // ✅ Baru ditambahkan
  rpc HealthCheck (google.protobuf.Empty) returns (google.protobuf.Empty);
}

message Complaint_serviceRequest {
  string user_id = 1;
  string input = 2;
}

message Complaint_serviceResponse {
  string status = 1;
  string result = 2;
}

message CreateComplaintRequest { // ✅ Baru
  string user_id = 1;
  string message = 2;
  string product = 3;
  string source = 4; // contoh: "chatbot", "web"
  string emotion = 5; // contoh: "disappointed"
}

message CreateComplaintResponse { // ✅ Baru
  string status = 1;
  string complaint_id = 2;
  string message = 3;
}
